CT-1722 Refactor SAR close to respond_and_close

stephenrichards opened the pull request:
SAR cases do not have uploaded responses, and so the first iteration was
that the responder simply closed the case when ready.  However, this meant
that the way of testing the response date of a case differed between FOI and
SAR cases, in that there was no respond event for SAR cases, and the close
event had to be used a a proxy.

This PR changes the close event for SARs to a respond_and_close which in fact
writes two events tothe transitions table: respond immediately followed by a
close

misaka commented on app/models/case.rb:
```diff
@@ -351,9 +351,8 @@ def responded?
-    response_event = transitions.responded.last
-    return false if response_event.nil?
-    response_event.created_at <= external_deadline
+    return false unless closed?
```
Wouldn't it be better to raise here so that we fail quickly?

stephenrichards committed:
CT-1722 Refactor SAR close to respond_and_close

SAR cases do not have uploaded responses, and so the first iteration was
that the responder simply closed the case when ready.  However, this meant
that the way of testing the response date of a case differed between FOI and
SAR cases, in that there was no respond event for SAR cases, and the close
event had to be used a a proxy.

This PR changes the close event for SARs to a respond_and_close which in fact
writes two events tothe transitions table: respond immediately followed by a
close

stephenrichards committed:
CT-1722 refactor sar policy for close and respond
